compiler/AST/stmt.cpp

```diff
@@ -600,6 +600,58 @@ BlockStmt::isCForLoop() const {
   return false;
 }
 
+static CallExpr* getMarker(BlockStmt* stmt, PrimitiveTag markerType) {
+  if (!stmt->body.empty()) {
+    if (auto call = toCallExpr(stmt->body.first())) {
+      if (call->isPrimitive(markerType)) {
+        return call;
+      }
+    }
+  }
+  return nullptr;
+}
+
+// See docs on PRIM_GPU_ATTRIBUTE_BLOCK and PRIM_GPU_PRIMITIVE_BLOCK
+// in primitive.cpp for what they do.
+
+bool BlockStmt::isGpuAttributeBlock() {
+  return getMarker(this, PRIM_GPU_ATTRIBUTE_BLOCK);
+}
+
+bool BlockStmt::isGpuPrimitivesBlock() {
+  return getMarker(this, PRIM_GPU_PRIMITIVE_BLOCK);
+}
+
+bool BlockStmt::isForGpuAttributes() {
```

Comment:
Logically, I cannot make the distinction between an "attribute block" and a "primitive block". I am wondering renaming this function to not have "attribute" in the name can help me.

Reply:
I've renamed it to `isGpuMetadata`. For clarity, I'll note the following:

* The 'attribute block' is an early-compilation structure meant to wrap the statement that needs to have GPU primitives applied to it.
* The 'primitive block' is just a container for primitive calls (which might also include normalization temps). It persists until late-stage compilation since we want to make sure primitives are moved together with their temps.